Source/WebCore/fileapi/Blob.cpp

```diff
@@ -184,6 +191,7 @@ Blob::Blob(ReferencingExistingBlobConstructor, ScriptExecutionContext* context,
     , m_size(blob.size())
     , m_memoryCost(blob.memoryCost())
     , m_internalURL(BlobURL::createInternalURL())
+    , m_topOrigin(context ? context->topOrigin().data() : SecurityOriginData::createOpaque())
```

Comment:
Ideally, we would not need this, but I found that caching the SecurityOriginData was needed here while I worked on #8258 because the ScriptExecutionContext could be destroyed before the Blob is destroyed (and we'll need it when we unregister). So, even though we don't need this additional data member in this patch, it will be helpful in a follow-up. I can change this like you described and add the data member in the follow up patch, though, if you prefer.

Reply:
I think this is good enough justification. I was just curious if we could do away with it or not.
Fine to include in this patch given the reason.